tests/conftest.py

```diff
@@ -44,6 +46,523 @@ def pytest_logger_config(logger_config):
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def project_factory(request):
+    """
+    Create a new project factory.
+    Calling this fixture creates new project.
+    """
+    instances = []
+
+    def factory(**kwargs):
+        """
+
+        Returns:
+            object: ocs_ci.ocs.resources.ocs instance of 'Project' kind.
+        """
+        if 'metadata' not in kwargs or 'namespace' not in kwargs.get(
+            'metadata'
+        ):
+            namespace = helpers.create_unique_resource_name(
+                'test',
+                'namespace'
+            )
+            kwargs['metadata'] = kwargs.get('metadata') or {}
+            kwargs['metadata']['namespace'] = namespace
+
+        proj_obj = OCS(
+            kind='Project',
+            **kwargs
+        )
+        assert proj_obj.ocp.new_project(namespace), (
+            f'Failed to create new project {namespace}'
+        )
+
+        instances.append(proj_obj)
+        return proj_obj
+
+    def finalizer():
+        """
+        Delete the Ceph block pool
+        """
+        for instance in instances:
+            if not instance.is_deleted:
+                ocp.switch_to_default_rook_cluster_project()
+                instance.ocp.delete(
+                    resource_name=instance.namespace
+                )
+                instance.ocp.wait_for_delete(
+                    instance.namespace
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def pvc_factory(request, rbd_storageclass_factory, project_factory):
+    """
+    Create a persistent Volume Claim factory. Calling this fixture creates new
+    PVC. For custom PVC provide 'storageclass' parameter.
+    """
+    instances = []
+    active_project = None
+
+    def factory(
+        project=None,
+        storageclass=None,
+        custom_data=None,
+        status=constants.STATUS_BOUND
+    ):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+            storageclass (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'StorageClass' kind.
+            custom_data (dict): If provided then PVC object is created
+                by using these data. Parameters `project` and `storageclass`
+                are not used but reference is set if provided.
+            status (str): If provided then factory waits for object to reach
+                desired state.
+
+        Returns:
+            object: helpers.create_pvc instance.
+        """
+        if custom_data:
+            pvc_obj = PVC(**custom_data)
+            pvc_obj.create(do_reload=False)
+        else:
+            nonlocal active_project
+            project = project or active_project or project_factory()
+            active_project = project
+            storageclass = storageclass or rbd_storageclass_factory()
```

Comment:
Will this create new storage class for each pvc?  Is this expected behavior to create for each pvc it's own storageclass? Or do we want to share one storage class for all of PVCs created in one object of pvc factory?

Reply:
It will create a new storage class per pvc through the `rbd_storageclass_factory` if one isn't provided, yes. If storageclass isn't something that needs to be created on a per pvc basis, we can always change the scope of the fixture to `class` instead of defaulting to `function`.

Reply:
I will create a similar construct to `active_project` so that if there is not provided custom storageclass and storageclass based on an interface exists then there will be not created new storageclass.